store/etcd/v3/etcd.go

```diff
@@ -469,7 +469,7 @@ func (l *etcdLock) Lock(stopChan chan struct{}) (<-chan struct{}, error) {
 		return nil, err
 	}
 
-	err = l.store.Put(l.writeKey, []byte(l.value), nil)
+	_, err = l.store.client.Put(ctx, l.writeKey, l.value, etcd.WithLease(l.session.Lease()))
```

Comment:
As explained in https://github.com/abronan/valkeyrie/issues/31#issuecomment-539899054 I think this should be an option rather than the default behavior, otherwise we fall back on being inconsistent by default with etcd deleting the key while other stores aren't.

Reply:
I'm not opposed to making it an option, but we're going to be inconsistent either way:
Redis and Dynamo are deleting the value, but not Consul or Zookeeper.

Reply:
Fixed!